src/components/CarListComponent/CarListComponent.tsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import {
+  Order_By,
+  Query_GetCarsQueryVariables,
+  useQuery_GetCarsQuery,
+} from '../../api/graphql/__generated__/graphql-types';
+import { isValidUuid } from '../../helpers/validators';
+import ErrorMessage from '../ErrorMessage/ErrorMessage';
+import FilterComponent from '../FilterComponent/FilterComponent';
+import HeaderList from '../HeaderList/HeaderList';
+import LoadingComponent from '../LoadingComponent/Loading';
+import BodyList from './BodyList/BodyList';
+import { Container } from './CarListComponentStyled';
+
+interface usercars {
+  user_id: number;
+  car_id: number;
+}
+
+interface CarListComponentProps {
+  favoritesCarsArray?: usercars[];
+}
+
+const CarListComponent = ({ favoritesCarsArray }: CarListComponentProps) => {
+  const { data, loading, error, refetch } = useQuery_GetCarsQuery();
+  const [searchParam, setSearchParam] = useSearchParams();
+  useEffect(() => {
+    if (!loading) {
+      if (isValidUuid(searchParam.get('search')!)) {

```

Comment:
Prefer a guard clause instead of `!`

`if (searchParam.get('search') === null) return`

Reply:
Understood Steven i will apply typeguards and avoid `!`